### ISSUE TITLE
Big multiplayer connection problem

### ISSUE THREAD

**Issue by MagicRS** [suriyun-production/bomber-io-docs#5]
Since recently without having touched a multiplayer configuration or a script etc.. As soon as someone connects to a running session, all the game bugs for them. 

- high latency
- No bombs
- Respawn button does not work

I have no idea where this can come from, Photon's configuration is good!
The game being online this is very problematic, do you have an idea? 

To try the game / problem , game : "BOMBERX"
You just have to be a player joining a game.

Thank you in advance!

![BUG1](https://user-images.githubusercontent.com/58114715/72027007-c5429f80-327d-11ea-9ea4-5912c26e2f24.PNG)

![BUG2](https://user-images.githubusercontent.com/58114715/72027233-7d704800-327e-11ea-8955-e1768cf379cb.PNG)

**Comment by insthync** [comment]
I can't find your game, I have some questions
* Is there any error in Console?
* There are how many bricks in the scene?
* There are how many bots in the scene?

**Comment by MagicRS** [comment]
![image](https://user-images.githubusercontent.com/58114715/72064647-9a3b6880-32dc-11ea-9ac1-27c32ba4fd28.png)


There are as many bricks as the original card and the number of bot can be from 1 to 9, it doesn't change.
I just noticed this problem that was not present during my testing phases.

Errors on the console are as follows:
**When a person joins the waiting room:**
![image](https://user-images.githubusercontent.com/58114715/72064198-a83cb980-32db-11ea-86e7-9eb11ffd6f13.png)

Errors are constantly on the rise (In Game :)
When the player invited to the game, leaves the game, the errors stop rising.
 
![image](https://user-images.githubusercontent.com/58114715/72064366-ffdb2500-32db-11ea-80d1-9446abd0c4ee.png)


![image](https://user-images.githubusercontent.com/58114715/72064406-171a1280-32dc-11ea-91c3-cdd0ab651f59.png)


![image](https://user-images.githubusercontent.com/58114715/72064428-213c1100-32dc-11ea-9841-ae7a7c1e414a.png)

I tried putting the original "gameplay / network and gamedata" scripts, but nothing works.
Try with the original (empty) map, problem persists.

**Comment by insthync** [comment]
Oh, you have to clear **transform** from bomb entities, powerup entities, character entities, bot entities.

![Screenshot_271](https://user-images.githubusercontent.com/2697550/72066974-7d228c00-3314-11ea-8576-311e7ee3b346.png)
![Screenshot_272](https://user-images.githubusercontent.com/2697550/72066975-7d228c00-3314-11ea-8bd8-08d7fc79ec30.png)
![Screenshot_273](https://user-images.githubusercontent.com/2697550/72066977-7d228c00-3314-11ea-864f-afd278047a7d.png)

**Comment by MagicRS** [comment]
What should I delete on the characters? :
![image](https://user-images.githubusercontent.com/58114715/72068449-5862f000-32e5-11ea-8fc2-168c310d7758.png)

The bots name no prefabs or I could delete a few things from them.

Thanks you

**Comment by MagicRS** [comment]
The multiplayer problem is not present when I play with an original version of the game but it is present on a version dated December 10, 2019 that worked at the time. 

Could it come from photon?
![image](https://user-images.githubusercontent.com/58114715/72070144-176cda80-32e9-11ea-9130-8144493ea1a9.png)

Reminder: 
Problem present for the guest. 
Visual teleportation of bots / enemies
Bug of bombs etc

**Comment by MagicRS** [comment]
The errors target the bricks, so I removed the PhotonView component target from the bricks and the errors are no longer there.
How do you explain such a situation all of a sudden?

However, connection bugs / latency with the incoming person persist.

![image](https://user-images.githubusercontent.com/58114715/72082141-f6fc4a80-32ff-11ea-9a24-c7725a8a3300.png)

**Comment by insthync** [comment]
Hmm, you don't have to remove PhotonView component from bricks, just remove transfrom from Observe Component setting.

**Comment by MagicRS** [comment]
Yes, yes, that's what I apologized for. So the mistakes are gone (which is good for the eyes) but the player bug remains. What blocks the multiplayer.

**the remaining errors are as follows:**
2 First are in waiting room et the last are In-Game steadily increasing
![image](https://user-images.githubusercontent.com/58114715/72090420-56615700-330e-11ea-9df1-2a648751b17b.png)

The error target "UINetwordGameScore.cs"
![image](https://user-images.githubusercontent.com/58114715/72090641-c079fc00-330e-11ea-8c4c-7b77b68baf3e.png)


Thank you for your help, I hope I won't have to solicit you as much afterwards ^^

**Comment by insthync** [comment]
Check your score Uis, any of `User Rankings` must not empty
![Screenshot_274](https://user-images.githubusercontent.com/2697550/72093760-a6104480-3347-11ea-9eef-fb2bd3f8f36a.png)

**Comment by MagicRS** [comment]
They are all present in all game modes (and well defined above).
![image](https://user-images.githubusercontent.com/58114715/72094523-e5726d00-3316-11ea-9e50-a62250c3da4f.png)
![image](https://user-images.githubusercontent.com/58114715/72094578-00dd7800-3317-11ea-8a0f-e16789272457.png)

This mistake is not a problem in itself, I don't see any consequences.

**Comment by insthync** [comment]
Hmm, weird, it should not be null, you may add code `if (userRanking)` at line 35 to avoid null exception.

**Comment by MagicRS** [comment]
I added the code and it's all good.

**When I start a game in offline mode, after clicking on "SRART GAME", the following error appears :**
![image](https://user-images.githubusercontent.com/58114715/72205119-af043180-347f-11ea-9647-76721c25b9ab.png)

Either I open a waiting room for no reason, or start a game without naming my character.
![image](https://user-images.githubusercontent.com/58114715/72205143-073b3380-3480-11ea-820a-049c6e4a5a74.png)


**Error Detail (Only in offline mode) :**

> Parameter name: key
> System.Collections.Generic.Dictionary`2[TKey,TValue].FindEntry (TKey key) (at <599589bf4ce248909b8a14cbe4a2034e>:0)
> System.Collections.Generic.Dictionary`2[TKey,TValue].ContainsKey (TKey key) (at <599589bf4ce248909b8a14cbe4a2034e>:0)
> UIPhotonWaitingRoom.DestroyPlayerUI (System.String id) (at Assets/BomberIO/SimplePhoton2NetworkManager/Scripts/UI/UIPhotonWaitingRoom.cs:219)
> UIPhotonWaitingRoom.CreatePlayerUI (Photon.Realtime.Player player) (at Assets/BomberIO/SimplePhoton2NetworkManager/Scripts/UI/UIPhotonWaitingRoom.cs:240)
> UIPhotonWaitingRoom.OnJoinedRoomCallback () (at Assets/BomberIO/SimplePhoton2NetworkManager/Scripts/UI/UIPhotonWaitingRoom.cs:203)
> UIPhotonWaitingRoom.Show () (at Assets/BomberIO/SimplePhoton2NetworkManager/Scripts/UI/UIPhotonWaitingRoom.cs:53)
> UIPhotonNetworking.OnJoinedRoomCallback () (at Assets/BomberIO/SimplePhoton2NetworkManager/Scripts/UI/UIPhotonNetworking.cs:107)
> SimplePhotonNetworkManager.OnJoinedRoom () (at Assets/BomberIO/SimplePhoton2NetworkManager/Scripts/Network/SimplePhotonNetworkManager.cs:534)
> BaseNetworkGameManager.OnJoinedRoom () (at 
> etc..


Target line in UPPhotonWaitingRoom.cs (219) :  if (waitingPlayers.ContainsKey(id))
SimplePhotonNetworkManager.cs (551) :  PhotonNetwork.JoinRandomRoom();
And many other line etc..

**Or if no error appears the game launches in error with no data :**
![image](https://user-images.githubusercontent.com/58114715/72205268-474ee600-3481-11ea-82b6-de3d8a0b5aa5.png)

**Comment by insthync** [comment]
[BomberIO-UpdateSimplePhotonNetworkManager.zip](https://github.com/suriyun-production/bomber-io-docs/files/4049725/BomberIO-UpdateSimplePhotonNetworkManager.zip)
Try import this package to fix it.

**Comment by MagicRS** [comment]
No more in-game errors and correct game start, thanks ! but still no player name in play

![](https://user-images.githubusercontent.com/58114715/72205143-073b3380-3480-11ea-820a-049c6e4a5a74.png)

**Comment by insthync** [comment]
It is occurring whennplaying offline?

**Comment by MagicRS** [comment]
Yes,

**Online mode (normal) :**
![image](https://user-images.githubusercontent.com/58114715/72269565-13f88c80-3624-11ea-970a-53f6247d41a2.png)
Console error : nothing

**Offline mode only :**
![image](https://user-images.githubusercontent.com/58114715/72269742-5e7a0900-3624-11ea-805e-bd0d192987c9.png)
Console error : nothing

This problem wasn't present before, it's as if the offline mode didn't read the playerprefs of the player name anymore.
